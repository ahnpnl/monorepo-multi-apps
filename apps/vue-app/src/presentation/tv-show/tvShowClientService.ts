import { TvShowService } from "@/core/component/tv-show/application/service/TvShowService.ts";
import { restApiClient } from "@/infrastructure/api-client/apiClient.ts";

export const tvShowClientService = new TvShowService(restApiClient);
