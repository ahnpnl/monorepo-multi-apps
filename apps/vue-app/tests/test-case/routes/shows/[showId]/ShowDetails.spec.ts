import { render } from "@testing-library/vue";
import { flushPromises } from "@vue/test-utils";

import { TestFixtureTvShowCastExtension } from "../../../../framework/extensions/fixtures/tv-shows/TestFixtureTvShowCastExtension.ts";
import { TestFixtureTvShowExtension } from "../../../../framework/extensions/fixtures/tv-shows/TestFixtureTvShowExtension.ts";
import { mockRouter } from "../../../../framework/test-double/router/mockVueRouter.ts";

import { tvShowClientService } from "@/presentation/tv-show/tvShowClientService.ts";
import ShowDetails from "@/routes/shows/[showId]/index.vue";

describe("ShowDetails", () => {
    describe("renders HTML", () => {
        it("should display show information and cast", async () => {
            const tvShowFixture = TestFixtureTvShowExtension.startNewFixture();
            tvShowFixture.withShows([
                {
                    id: 5,
                    name: "True Detective",
                    language: "English",
                    genres: ["Drama", "Crime", "Thriller"],
                    status: "Running",
                    runtime: 60,
                    premiered: "2014-01-12",
                    image: {
                        medium: "https://static.tvmaze.com/uploads/images/medium_portrait/490/1226764.jpg",
                        original: "https://static.tvmaze.com/uploads/images/original_untouched/490/1226764.jpg",
                    },
                    summary:
                        "<p>Touch darkness and darkness touches you back. <b>True Detective</b> centers on troubled cops and the investigations that drive them to the edge. Each season features a new cast and a new case.</p><p><i><b>True Detective</b></i> is an American anthology crime drama television series created and written by Nic Pizzolatto.Â </p>",
                },
            ]);
            const tvShowDetailsStub = tvShowFixture.toJson()[0];
            const tvShowCastFixture = TestFixtureTvShowCastExtension.startNewFixture();
            tvShowCastFixture.withCasts([
                {
                    person: {
                        id: 559,
                        url: "https://www.tvmaze.com/characters/559/true-detective-maggie-hart",
                        name: "Maggie Hart",
                        image: {
                            medium: "https://static.tvmaze.com/uploads/images/medium_portrait/0/62.jpg",
                            original: "https://static.tvmaze.com/uploads/images/original_untouched/0/62.jpg",
                        },
                    },
                },
            ]);
            const tvShowCastsStub = tvShowCastFixture.toJson();
            tvShowClientService.queryShowDetail = vi.fn().mockResolvedValueOnce(tvShowDetailsStub);
            tvShowClientService.queryShowCasts = vi.fn().mockResolvedValueOnce(tvShowCastsStub);
            const wrapper = render(ShowDetails, {
                global: {
                    plugins: [mockRouter],
                },
            });
            await flushPromises();

            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
