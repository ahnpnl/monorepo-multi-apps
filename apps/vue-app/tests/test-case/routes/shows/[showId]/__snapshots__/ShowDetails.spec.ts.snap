// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShowDetails > renders HTML > should display show information and cast 1`] = `
"<div data-v-572bbe56="" class="tv-show-details">
  <div data-v-572bbe56="" class="tv-show-details__information flex md-flex-row gap-10 items-center"><img data-v-572bbe56="" src="https://static.tvmaze.com/uploads/images/original_untouched/490/1226764.jpg" alt="True Detective" width="400" height="600">
    <div data-v-572bbe56="" class="details-information flex flex-col gap-10">
      <h2 data-v-572bbe56="">Storyline</h2>
      <p data-v-572bbe56=""> Touch darkness and darkness touches you back. True Detective centers on troubled cops and the investigations that drive them to the edge. Each season features a new cast and a new case. True Detective is an American anthology crime drama television series created and written by Nic Pizzolatto.&nbsp; </p>
      <div data-v-572bbe56="" class="details-information__side-information grid gap-4 items-center"><span data-v-572bbe56="">Released</span><span data-v-572bbe56="">2014-01-12</span><span data-v-572bbe56="">Runtime</span><span data-v-572bbe56="">1h</span><span data-v-572bbe56="">Genre</span><span data-v-572bbe56="">Drama, Crime, Thriller</span><span data-v-572bbe56="">Status</span><span data-v-572bbe56="">Running</span><span data-v-572bbe56="">Language</span><span data-v-572bbe56="">English</span></div>
    </div>
  </div>
  <div data-v-572bbe56="" class="tv-show-details__cast">
    <h2 data-v-572bbe56="">Cast</h2>
    <div data-v-572bbe56="" class="carousel-items">
      <div data-v-572bbe56="" class="carousel-items__item flex flex-row gap-2">
        <div data-v-8024ae0a="" data-v-572bbe56="" class="media-card" data-testid="media-card-559"><a data-v-8024ae0a="" class="block" href="https://www.tvmaze.com/characters/559/true-detective-maggie-hart">
            <div data-v-8024ae0a="" class="media-card__picture">
              <picture data-v-8024ae0a=""><img data-v-8024ae0a="" src="https://static.tvmaze.com/uploads/images/medium_portrait/0/62.jpg" alt="Maggie Hart" height="270" width="185"></picture>
            </div><span data-v-8024ae0a="">Maggie Hart</span>
          </a></div>
      </div>
    </div>
  </div>
</div>"
`;
